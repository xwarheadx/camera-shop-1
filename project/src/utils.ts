import { PAGE_LIMIT } from './consts';
import { reviewDataType } from '../src/types/review';

export function getPagesCount(cameraCount: number): number {
  return Math.ceil(cameraCount / PAGE_LIMIT);
}

export function getNumeric(page: string) {
  return Number(page.replace( /^\D+/g, ''));
}

export function seperatePrice(priceValue: number) {
  let returnedPrice = String(priceValue);
  if (returnedPrice.length > 3) {
    returnedPrice = `${returnedPrice.slice(0,2)} ${returnedPrice.slice(2)}`;
  }
  return returnedPrice;
}

export function validateInput(target: (EventTarget & HTMLInputElement) | (EventTarget & HTMLTextAreaElement)) {
  if (target.parentElement?.parentElement && target.name === 'rating' && (Number(target.value) > 0)) {
    if (!target.parentElement.parentElement.parentElement?.className.includes('is-valid')) {
      target.parentElement.parentElement.parentElement?.classList.add('is-valid');
      target.parentElement.parentElement.parentElement?.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && target.name !== 'review' && typeof(target.value) === 'string') {
    if (!target.parentElement.parentElement.className.includes('is-valid') && target.value.length > 0) {
      target.parentElement.parentElement.className += ' is-valid';
      target.parentElement.parentElement.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && typeof(target.value) === 'string'){
    if(target.value.length === 0) {
      target.parentElement.parentElement.className += ' is-invalid';
      target.parentElement.parentElement.classList.remove('is-valid');
    }
  }

  if (target.parentElement?.parentElement && typeof(target.value) === 'string') {
    if (!target.parentElement.parentElement.className.includes('is-valid') && target.value.length >= 5) {
      target.parentElement.parentElement.className += ' is-valid';
      target.parentElement.parentElement.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && target.name === 'review' && target.value.length < 5) {
    if (!target.parentElement.parentElement.className.includes('is-invalid')) {
      target.parentElement.parentElement.className += ' is-invalid';
      target.parentElement.parentElement.classList.remove('is-valid');
    }
  }

}
export function validateForm(data: reviewDataType) {
  let validationStatus = true;
  Object.entries(data).forEach(([key, value]) =>{
    if (key !== 'review') {
      if(value === '') {
        const failedInput = document.querySelector(`input[name=${key}]`);
        validateInput(failedInput as HTMLInputElement);
        validationStatus = false;
      }
    }
    if(key === 'review') {
      if(String(value).length < 5) {
        const failedInput = document.querySelector(`textarea[name=${key}]`);
        validateInput(failedInput as HTMLInputElement);
      }
    }
    if(key === 'rating') {
      if(value < 1) {
        const failedInput = document.querySelector(`input[name=${key}]`);
        if(failedInput && failedInput.parentElement?.parentElement){
          failedInput.parentElement.parentElement.parentElement?.classList.add('is-invalid');
        }
        validationStatus = false;
      }
    }
  });
  return validationStatus;
}


