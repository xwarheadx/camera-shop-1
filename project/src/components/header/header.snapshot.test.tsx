import { renderWithProviders } from '../../test/utils/render-with-redux';
import Header from './header';

test('Header renders correctly', () => {
  const { asFragment } = renderWithProviders(<Header/>);
  expect(asFragment()).toMatchSnapshot();
});
