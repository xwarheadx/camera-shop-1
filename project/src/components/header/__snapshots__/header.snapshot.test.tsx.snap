// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<DocumentFragment>
  <header
    class="header"
    id="header"
  >
    <div
      class="container"
    >
      <a
        aria-label="Переход на главную"
        class="header__logo"
        href="#!"
      >
        <svg
          aria-hidden="true"
          height="36"
          width="100"
        >
          <use
            xlink:href="#icon-logo"
          />
        </svg>
      </a>
      <nav
        class="main-nav header__main-nav"
      >
        <ul
          class="main-nav__list"
        >
          <li
            class="main-nav__item"
          >
            <a
              class="main-nav__link"
              href="/catalog"
            >
              Каталог
            </a>
          </li>
          <li
            class="main-nav__item"
          >
            <a
              class="main-nav__link"
              href="/"
            >
              Гарантии
            </a>
          </li>
          <li
            class="main-nav__item"
          >
            <a
              class="main-nav__link"
              href="/"
            >
              Доставка
            </a>
          </li>
          <li
            class="main-nav__item"
          >
            <a
              class="main-nav__link"
              href="/"
            >
              О компании
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="form-search"
      >
        <form>
          <label>
            <svg
              aria-hidden="true"
              class="form-search__icon"
              height="16"
              width="16"
            >
              <use
                xlink:href="#icon-lens"
              />
            </svg>
            <input
              autocomplete="off"
              class="form-search__input"
              placeholder="Поиск по сайту"
              type="text"
            />
          </label>
          <ul
            class="form-search__select-list"
          >
            <li
              class="form-search__select-item"
            >
              Cannonball Pro MX 8i
            </li>
            <li
              class="form-search__select-item"
            >
              Cannonball Pro MX 7i
            </li>
            <li
              class="form-search__select-item"
            >
              Cannonball Pro MX 6i
            </li>
            <li
              class="form-search__select-item"
            >
              Cannonball Pro MX 5i
            </li>
            <li
              class="form-search__select-item"
            >
              Cannonball Pro MX 4i
            </li>
          </ul>
        </form>
        <button
          class="form-search__reset"
          type="reset"
        >
          <svg
            aria-hidden="true"
            height="10"
            width="10"
          >
            <use
              xlink:href="#icon-close"
            />
          </svg>
          <span
            class="visually-hidden"
          >
            Сбросить поиск
          </span>
        </button>
      </div>
      <a
        class="header__basket-link"
        href="/"
      >
        <svg
          aria-hidden="true"
          height="16"
          width="16"
        >
          <use
            xlink:href="#icon-basket"
          />
        </svg>
      </a>
    </div>
  </header>
</DocumentFragment>
`;
